### DISCUSSION TITLE
Town building animation speed

### DISCUSSION THREAD

**Discussion by shaman4d** [ihhub/fheroes2#5862] [Ideas]
During playing the game I noticed that I feel very annoyed when build some building in a town and waiting when it complete to buy new units. I want to add experimental option: UI: building animation double speed, to speed up animation. 
What do you think?

**Comment by Branikolog** [comment]
Hi, @shaman4d .
I'm glad you're interested in our project and improving Heroes 2 game.
We want to make fheroes2 engine as much usable and friendly, as possible. That's why we're aware of adding numerous debatable experimental options. 
Here you can read a short remark about it:
https://github.com/ihhub/fheroes2/wiki/Adding-a-new-option-into-the-game

To the topic. I've experienced the same inconvenience while evolving castles in game. Not sure, we have an opened issue for that right now. But there was an opened PR to fix the issue, but we didn't come out with the proper solution. Speeding default animations will look not so good, so we cannot afford it even as experimental option. We need to implement a solution, which wouldn't spoil game style being as close to the original operation, as possible. I think the solution is going to be quite complex. Such situations as inspecting a hero, re-arrangement of troops, hiring troops in a dwellings and getting out of the castle screen should be discussed and have personal approach within the PR.

**Reply by zenseii** [reply]
Hi guys,

I believe the original game (OG) lets you interrupt these animations by opening something else like exiting the castle or opening any new window etc. This is just a suggestion for one way to approach this issue that would be better than just 2x speed because that could really mess with the sound of that animation too (chipmunks voices?).

**Comment by LeHerosInconnu** [comment]
Hello @shaman4d, @Branikolog and @zenseii,

It's a annoyance that I feel and that is also present in the original game.
Also, in the original game, the player's click is "memorized" and is taken into account at the end of the animation.
We should improve this in fheroes2 and make sure that the user always has priority when he performs an action and not wait as it is the case now for the animations.
Why not just go directly to the end of the animation (the last frame) when the user performs an action (click on the EXIT button, etc.).
